sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/PipelinedDocumentQueryParams.java

```diff
@@ -52,6 +55,7 @@ public PipelinedDocumentQueryParams(
         this.cosmosQueryRequestOptions = cosmosQueryRequestOptions;
         this.correlatedActivityId = correlatedActivityId;
         this.feedRanges = feedRanges;
+        this.isQueryCancelledOnTimeout = isQueryCancelledOnTimeout;
```

Comment:
I notice, we are using the same instance of isQueryCancelledTimout throughout the SDK, is it intentional? If so, multiple places can update it and it will all reflect everywhere, starting with RxDocumentClientImpl.

Reply:
Yes, this is intentional. `this.isQueryCancelledOnTimeout` reference is immutable but the value it wraps should reflect everywhere in the query pipeline.